internal/pkg/manifest/interpolate.go

```diff
@@ -72,7 +72,21 @@ func (i *Interpolator) applyInterpolation(node *yaml.Node) error {
 		if err != nil {
 			return err
 		}
-		node.Value = interpolated
+		if strings.HasPrefix(interpolated, "[") && strings.HasSuffix(interpolated, "]") {
```

Comment:
I wonder if this is more robust:
```suggestion
                var s []string
                err := json.Unmarshal([]byte(interpolated), &s)
                if err != nil && len(s) != 0 {
                     // `interpolated` is a slice, do the node type conversion
                }
```